### PR TITLE
Possible fix for 119 deployments

### PR DESCRIPTION
**Pull request by eanders**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

This includes a security update for rails and fix for indexing a table that was breaking deploys

## Type of change
[//]: # 'remove options that are not relevant'
- [x] Bug fix
- [x] Dependency update

## Checklist before requesting review
- [ ] I have performed a self-review of my code
- [ ] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [ ] My code follows the style guidelines of this project (rubocop)
- [ ] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Comment by eanders** [comment]
@ttoomey any thoughts on why the gem update might have [broken these tests](https://github.com/greenriver/hmis-warehouse/actions/runs/9393403570/job/25869394591)?

**Comment by ttoomey** [comment]
> @ttoomey any thoughts on why the gem update might have [broken these tests](https://github.com/greenriver/hmis-warehouse/actions/runs/9393403570/job/25869394591)?

@eanders I think you must have run the view migration in 4818 during code review. There was an error in the view that was fixed before it got merged. But it looks like your db retained that error and it's getting pushed up here

Can you try rerunning it and pushing up the new schema?
```
RAILS_ENV=test rails db:migrate:redo:warehouse VERSION=20240526045112
```

Our `db/warehouse_structure.sql` has so much churn. There must be a better solution

**Comment by eanders** [comment]
That was it, thank you, and yes, there must be a better way!

File: drivers/hmis/app/models/hmis/hud/concerns/has_custom_data_elements.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
 ###
 
-module Hmis::Hud::Concerns::FormSubmittable
+module Hmis::Hud::Concerns::HasCustomDataElements
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/hud/concerns/has_custom_data_elements.rb]: *edit - this file should have been deleted, sorry. removed

**Review by gigxz — APPROVED** [review]

**Pull request merged**